File: src/simplifyDirectoryPath.java
import java.util.Stack;

public class Solution{
    public String simplifyPath(String A) {
        Stack<String> stack = new Stack<>();
        String[] arr = A.split("/");
        for(String s: arr){
            if(s.equals("..")){
                if(!stack.isEmpty()) stack.pop();
            }
            else if(s.equals(".") || s.equals("")) continue;
            else stack.push(s);
        }
        return "/"+String.join("/", stack);
    }
}

File: src/simplifyDirectoryPath.py
class Solution:
	# @param A : string
	# @return a strings
	def simplifyPath(self, A):
        st = []
        for i in A.split("/"):
            if i == "..":
                if st:
                    st.pop()
            elif i == "." or i == "":
                continue
            else:
                st.append(i)
        return "/" + "/".join(st)